cmd/tracee-ebpf/internal/printer/printer.go

```diff
@@ -29,51 +29,64 @@ type eventPrinter interface {
 	Close()
 }
 
-func newEventPrinter(kind string, containerMode bool, relativeTS bool, out io.WriteCloser, err io.WriteCloser) (eventPrinter, error) {
+type Config struct {
+	Kind          string
+	OutPath       string
+	OutFile       io.WriteCloser
+	ErrPath       string
+	ErrFile       io.WriteCloser
+	ContainerMode bool
+	RelativeTS    bool
+	Protocol      bool
+}
+
+func New(config Config) (eventPrinter, error) {
 	var res eventPrinter
-	var initError error
+	kind := config.Kind
 	switch {
```

Comment:
```suggestion
	switch config.Kind {
	case "ignore":
	....
```

Reply:
I would still like to keep it in a variable for later use in printing and such, so this doesn't really make the code cleaner later IMO.

Reply:
If we use the switch the way you suggest I need to check for the gotemplate path first since the `strings.HasPrefix` can't be compared to an actual string, so I would have to check it in an `if` statement first, and then have the actual `kind` switch in an else statement... 
I understand wanting to clean up the switch but there's no way to actually do so without getting rid of the gotemplate path (which would basically be like getting rid of the feature for no reason), otherwise we get even worse code IMO.